src/physics/base/GeantImporter.hh

```diff
@@ -0,0 +1,111 @@
+//---------------------------------------------------------------------------//
+//! \file GeantImporter.hh
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Loads the geant-exporter-data.root file data into memory
+ *
+ * GeantImporter loads particle data and physics tables previously saved into
+ * a ROOT file by the app/geant-exporter external code.
+ *
+ * Usage:
+ * \code
+ *  GeantImporter importer;
+ *  importer.load_root_file("/path/to/file.root");
+ *  importer.load_particle_data();
+ *  importer.load_physics_tables();
+ * \endcode
+ *
+ * These methods will load into memory both particle and physics table data
+ * as a vector<GeantParticle> and a vector<GeantPhysicsTable>. The data can
+ * be retrieved and copied to a GeantParticle or GeantPhysicsTable via copy
+ * functions:
+ * \code
+ *  importer.get_GeantParticle(...);
+ *  importer.get_GeantPhysicsTable(...);
+ * \endcode
+ * 
+ * The protected vector<GeantParticle> and vector<GeantPhysicsTable> members
+ * are also available via
+ * \code
+ *  importer.get_vector_GeantParticle();
+ *  importer.get_vector_GeantPhysicsTable();
+ * \endcode
+ * 
+ * A ParticleParams shared pointer is provided by
+ * \code
+ *  importer.get_ParticleParams()
+ * \endcode
+ * 
+ * Lastly, if a direct ParticleDef or vector<ParticleDef> object are needed, 
+ * both can be created using
+ * \code
+ *  importer.particleDef(PDGNumber pdg);
+ *  importer particleDef(GeantParticle& particle);
+ *  importer.particleDef_vector();
+ * \endcode
+ */
+//-------------------------------------------------------------------------//
+
+#pragma once
+
+// C++
+#include <memory>
+#include <string>
+#include <vector>
+
+// Project
+#include "GeantParticle.hh"
+#include "ParticleParams.hh"
+#include "ParticleDef.hh"
+
+#include "base/Types.hh"
+#include "base/Macros.hh"
+
+// ROOT
+class TFile;
+
+namespace celeritas
+{
+//-------------------------------------------------------------------------//
+
+class GeantImporter
+{
+  private:
+    std::unique_ptr<TFile>         root_input_;
+    std::vector<std::string>       object_list_;
+    std::vector<GeantParticle>     particle_vector_;
+
+    void build_object_list(std::string const root_folder);
+
+  public:
+    GeantImporter();
+    ~GeantImporter();
+
+    // Loads the ROOT data into memeory
+    void load_root_file(std::string const filename);
+
+    void load_particle_data();
+
+    std::vector<GeantParticle>     get_vector_GeantParticle();
+
+    // Print capabilities
+    void print_object_list();
+
+    // Copy the data from memory into an object
+    bool get_GeantParticle(PDGNumber pdg, GeantParticle& particle);
+
+    // Create ParticleDef
+    ParticleDef              particleDef(PDGNumber pdg);
+    ParticleDef              particleDef(GeantParticle& particle);
+    std::vector<ParticleDef> particleDef_vector();
+
+    // Create ParticleParams
+    std::shared_ptr<ParticleParams> get_ParticleParams();
```

Comment:
Most of the methods here are redundant (though it is probably a good idea to keep some of these as implementation helper functions).